web/modules/custom/joinup_notification/joinup_notification.behat.inc

```diff
@@ -55,6 +55,7 @@ class JoinupNotificationSubContext extends DrupalSubContextBase implements Drupa
       throw new \Exception("User {$recipient} does not have an email.");
     }
 
+    \Drupal::state()->resetCache();
```

Comment:
I've seen that it works also if moved in the `beforeEmailScenario()`, just after setting the test mail collector.